lua/nvim_lsp/rust_analyzer.lua

```diff
@@ -1,11 +1,88 @@
 local configs = require 'nvim_lsp/configs'
 local util = require 'nvim_lsp/util'
+local server_name = 'rust_analyzer'
+local release_url = 'https://api.github.com/repos/rust-analyzer/rust-analyzer/releases/latest'
+local postfix = '-linux'
+local bin = 'rust-analyzer'
+
+if vim.fn.has('mac') == 1 then
+  postfix = '-mac'
+elseif vim.fn.has('win32') == 1  or vim.fn.has('win64') == 1 then
+  postfix = '-windows.exe'
+end
+
+local function make_installer()
+  local I = {}
+  local P = util.path.join
+  local install_dir = P{util.base_install_dir, server_name}
+	local cmd_path = P{install_dir, bin}
+
+  local function getReleaseJson()
```

Comment:
Use snake_case. I'm not convinced it's a good name either since json is just an intermediate format, it doesn't return it. Maybe `get_release_info()`?